utility/src/main/java/org/oppia/util/parser/CustomBulletSpan.kt

```diff
@@ -48,37 +48,48 @@ class CustomBulletSpan(context: Context) : LeadingMarginSpan {
     start: Int,
     end: Int,
     first: Boolean,
-    layout: Layout?
+    layout: Layout
   ) {
-    if ((text as Spanned).getSpanStart(this) == start) {
-      val style = paint.style
-      paint.style = Paint.Style.FILL
+    val startCharOfSpan = (text as Spanned).getSpanStart(this)
+    val isFirstCharacter = startCharOfSpan == start
 
-      var yPosition = if (layout != null) {
-        val line = layout.getLineForOffset(start)
-        layout.getLineBaseline(line).toFloat() - bulletRadius * 2f
+    if (isFirstCharacter) {
+      // Depending on the phone, x might always be 0. We need to re-calculate it here.
+      val trueX = 24f + gapWidth * indentation
+      if (string != "•") {
```

Comment:
Instead of putting a sentinel string value here, could we set up the custom span to always draw a bullet if it's a bullet? If I understand the current code CustomBulletSpan is also being used for numbered lists which I don't think we want. Could we add another custom span that just handles the leading logic for numbered lists & keep this one specialized for bullet lists?

Reply:
Separated custom span for bullet and numbered list.